### PR TITLE
H-694: Enable additional auto-updates in Renovate

### PR DESCRIPTION
**Pull request by nonparibus**
## 🌟 What is the purpose of this PR?

Applies @TimDiekmann's suggestions from [chat](https://hashintel.slack.com/archives/C02TWBTT3ED/p1693568163092989?thread_ts=1693567980.352119&cid=C02TWBTT3ED) (_internal link_)

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -51,7 +52,7 @@
     },
     {
       "matchDepTypes": ["devDependencies"],
-      "matchPackagePatterns": ["^jest-", "jest$", "^prettier", "prettier$"],
+      "matchPackagePatterns": ["^jest-", "jest$", "^prettier", "prettier$", "lint-staged"],
```
Comment (nonparibus) [inline .github/renovate.json:55]: Realized we could include this in addition.

**Review by TimDiekmann — DISMISSED** [review]

File: .github/renovate.json (hunk 1 above)
Comment (TimDiekmann) [reply .github/renovate.json:55]: Yup, good spot!

Comment (nonparibus) [inline .github/renovate.json:55]:
```suggestion
      "matchPackagePatterns": ["^jest-", "jest$", "lint-staged", "markdownlint-cli", "^prettier", "prettier$"],
```

Comment (TimDiekmann) [reply .github/renovate.json:55]:
```suggestion
      "matchPackagePatterns": ["^jest-", "jest$", "lint-staged", "lock file-lint", "markdownlint-cli", "^prettier", "prettier$"],
```

@TimDiekmann Sorry, one more... and alphabetized.

File: .github/renovate.json [hunk 2]
```diff
@@ -51,7 +52,7 @@
     },
     {
       "matchDepTypes": ["devDependencies"],
-      "matchPackagePatterns": ["^jest-", "jest$", "^prettier", "prettier$"],
+      "matchPackagePatterns": ["^jest-", "jest$", "lint-staged", "markdownlint-cli", "^prettier", "prettier$"],
```
Comment (nonparibus) [inline .github/renovate.json:55]:
```suggestion
      "matchPackagePatterns": ["^jest-", "jest$", "lint-staged", "lockfile-lint", "markdownlint-cli", "^prettier", "prettier$"],
```

Suggested by @TimDiekmann in https://github.com/hashintel/hash/pull/3050#discussion_r1312959189

File: .github/renovate.json (hunk 1 above)
Comment (nonparibus) [reply .github/renovate.json]: Addressed in https://github.com/hashintel/hash/pull/3050/commits/48b25719756dd2e50f3c83c6412871e479586bdd

**Review by TimDiekmann — APPROVED** [review]

**Pull request merged**